pkgs/development/libraries/gnutls/ssl-cert-file.patch

```diff
@@ -0,0 +1,36 @@
+From 53091092876e668a4c43a4944d1b821015dea7a3 Mon Sep 17 00:00:00 2001
+From: Yegor Timoshenko <yegortimoshenko@riseup.net>
+Date: Wed, 17 Oct 2018 07:48:34 +0000
+Subject: [PATCH] Handle SSL_CERT_FILE environment variable
+
+---
+ lib/system/certs.c | 13 +++++++++++++
+ 1 file changed, 13 insertions(+)
+
+diff --git a/lib/system/certs.c b/lib/system/certs.c
+index 53eb561d0..6adb960e3 100644
+--- a/lib/system/certs.c
++++ b/lib/system/certs.c
+@@ -137,6 +137,19 @@ add_system_trust(gnutls_x509_trust_list_t list,
+ 		r += ret;
+ #endif
+ 
++	char *env = secure_getenv("SSL_CERT_FILE");
```

Comment:
Can you ask upstream to incorporate this patch? Even if they refuse to merge it, we still get a code review from them.

Reply:
Yeah, sure, I'll try. However, to make it upstreamable it should probably be under an option.

The patch is trivial though, this chunk looks exactly like `DEFAULT_TRUST_STORE_FILE` above it, other than `secure_getenv` call.

Reply:
Why would it need to be behind a configure option?

Reply:
From what I gather, upstream has been hostile to including this previously. So I have little faith for it to be included without it being explicitly enabled by a configure flag.

Reply:
Where was this proposed before?

Reply:
https://guix-devel.gnu.narkive.com/aUu3aCUl/patch-gnu-gnutls-configure-location-of-system-wide-trust-store

Reply:
[This](https://lists.gnupg.org/pipermail/gnutls-devel/2015-February/007442.html) is the actual upstream thread where the guix developers talked to gnutls. The gnutls people don't really argue against the patch, they just mention that they think [using pkcs11](https://fedoraproject.org/wiki/Features/SharedSystemCertificates) is a better idea. Not sure if that is applicable for us, if I understand that correctly that needs a central service (hard on non-nixos). But upstream never really said "no that patch is a no-go because...".